src/plugins/mer/merhardwaredevicewizardpages.cpp

```diff
@@ -148,11 +153,32 @@ void MerHardwareDeviceWizardSelectionPage::handleTestConnectionClicked()
         goto end;
     }
 
+    m_ui->connectionTestLabel->setText(tr("Detecting SDK tools..."));
+    m_sdkClientToolsInstalled = detectSdkClientTools(sshParams, &m_connectionTestOk);
+    if (!m_connectionTestOk) {
+        m_ui->connectionTestLabel->setText(tr("Could not autodetect SDK tools"));
+        goto end;
+    }
+    if (!m_sdkClientToolsInstalled) {
+        m_ui->connectionTestLabel->setText(tr("SDK tools are not installed"));
+        if (askInstallSdkClientTools(password)) {
+            m_ui->connectionTestLabel->setText(tr("Installing SDK tools..."));
+            m_sdkClientToolsInstalled = installSdkClientTools(sshParams, password);
+            if (!m_sdkClientToolsInstalled) {
+                m_ui->connectionTestLabel->setText(tr("SDK tools install failed"));
+                goto end;
+            }
+        } else {
+            goto end;
+        }
+    }
+
     m_ui->connectionTestLabel->setText(tr("Connected %1 device").arg(m_deviceName));
 
 end:
     m_ui->testButton->setEnabled(true);
     m_isIdle = true;
+    password.truncate(0);
```

Comment:
password is about to go out of scope, what's the point of truncate?

Reply:
It was meant to be clear() but anyway looking at qt code calling clear doesn't do anything that destroying QString wouldn't do, so I'll just remove these.

In devicelock code values containing security code are always cleared so I did that here too but that's not really useful.